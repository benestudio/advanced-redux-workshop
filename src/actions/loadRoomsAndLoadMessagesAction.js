import loadRoomsAction from './loadRoomsAction';
import loadMessagesAction from './loadMessagesAction';
import selectRoomAction from './selectRoomAction';

export default (index = 0) => async dispatch => {
  await dispatch(loadRoomsAction());
  await dispatch(selectRoomAction(index));
  await dispatch(loadMessagesAction());
};
