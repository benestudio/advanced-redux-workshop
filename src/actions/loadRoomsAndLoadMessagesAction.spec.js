jest.mock('../api/loadMessages');
jest.mock('../api/loadRooms');

import loadRoomsAndLoadMessagesAction from './loadRoomsAndLoadMessagesAction';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

describe('loadRoomsAndLoadMessagesAction', () => {
  it('is a function', () => {
    expect(loadRoomsAndLoadMessagesAction.constructor).toBe(Function);
  });

  it('should invoke actions correctly', () => {
    const store = mockStore({ rooms: {}, messages: {} });

    return store.dispatch(loadRoomsAndLoadMessagesAction(0)).then(() => {
      // return of async actions
      expect(store.getActions()).toMatchSnapshot();
    })
  });
});
