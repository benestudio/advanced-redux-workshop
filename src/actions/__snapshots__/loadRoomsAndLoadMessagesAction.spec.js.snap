// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadRoomsAndLoadMessagesAction should invoke actions correctly 1`] = `
Array [
  Object {
    "type": "ROOMS_FETCH_START",
  },
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "id": "1",
          "participants": Array [
            "Alice",
            "Charlie",
          ],
        },
        Object {
          "id": "2",
          "participants": Array [
            "Bob",
            "Charlie",
          ],
        },
        Object {
          "id": "3",
          "participants": Array [
            "Alice",
            "Bob",
            "Charlie",
          ],
        },
      ],
    },
    "type": "ROOMS_FETCH_SUCCESS",
  },
  Object {
    "payload": Object {
      "index": 0,
    },
    "type": "ROOMS_SELECT",
  },
  Object {
    "type": "MESSAGES_FETCH_START",
  },
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "created": "1532432201308",
          "id": "1",
          "message": "Hi Bob!",
          "sender": "Alice",
        },
        Object {
          "created": "1532432212345",
          "id": "2",
          "message": "Hi Alice!",
          "sender": "Bob",
        },
        Object {
          "created": "1532432501308",
          "id": "3",
          "message": "Hi Bob!",
          "sender": "Alice",
        },
        Object {
          "created": "1532432812345",
          "id": "4",
          "message": "Hi Alice!",
          "sender": "Bob",
        },
      ],
    },
    "type": "MESSAGES_FETCH_SUCCESS",
  },
]
`;
